source/api_cc/src/DeepPotPT.cc

```diff
@@ -53,8 +53,13 @@ void DeepPotPT::init(const std::string& model,
     std::cout << "load model from: " << model << " to gpu " << gpu_id
               << std::endl;
   }
-  module = torch::jit::load(model, device);
-
+  std::unordered_map<std::string, std::string> metadata = {{"type", ""}};
+  module = torch::jit::load(model, device, metadata);
+  if (metadata["type"] == "dpa2") {
+    model_type = 1;
+  } else {
+    model_type = 0;
+  }
```

Comment:
I would suggest using the metadata provide the key 
`metadata["message_passing"] = True/False`
to distinguish if the model needs to handle the communications. 

The reason to change the name of the key is to make the variable more self-explanatory. 

I also discourage the variable name `model_type`. I would suggest something like `do_message_passing`.